### PR TITLE
[Refactor] Convert append methods in ZMConversation

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Converted append methods to Swift and replaced ZMGenericMessage with GenericMessage.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation+Message.swift [hunk 1]
```diff
@@ -206,9 +226,9 @@ extension ZMConversation {
     
     // MARK: - Helper methods
     
-    func append(message: MessageContentType, nonce: UUID = UUID(), hidden: Bool = false, expires: Bool = false) -> ZMClientMessage? {
-        return appendClientMessage(with: ZMGenericMessage.message(content: message, nonce: nonce), expires: expires, hidden: hidden)
-    }
+//    func append(message: MessageContentType, nonce: UUID = UUID(), hidden: Bool = false, expires: Bool = false) -> ZMClientMessage? {
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation+Message.swift:229]: Should be deleted?

File: Source/Model/Conversation/ZMConversation.m [hunk 2]
```diff
@@ -768,16 +734,16 @@ - (void)unarchiveIfNeeded
...
-{
-    VerifyReturnNil(genericMessage != nil);
-
-    ZMConversation *selfConversation = [ZMConversation selfConversationInContext:moc];
-    VerifyReturnNil(selfConversation != nil);
-    
-    ZMClientMessage *clientMessage = [selfConversation appendClientMessageWithGenericMessage:genericMessage expires:NO hidden:NO];
-    return clientMessage;
-}
+//+ (ZMClientMessage *)appendSelfConversationWithGenericMessage:(ZMGenericMessage * )genericMessage managedObjectContext:(NSManagedObjectContext *)moc;
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation.m:737]: Should be deleted?

File: Source/Model/Conversation/ZMConversation.m [hunk 3]
```diff
@@ -793,20 +759,20 @@ + (void)updateConversationWithZMLastReadFromSelfConversation:(ZMLastRead *)lastR
...
-        return nil;
-    }
-    
-    NSUUID *nonce = [NSUUID UUID];
-    ZMGenericMessage *message = [ZMGenericMessage messageWithContent:[ZMCleared clearedWithTimestamp:cleared conversationRemoteID:convID] nonce:nonce];
-    VerifyReturnNil(message != nil);
-    
-    return [self appendSelfConversationWithGenericMessage:message managedObjectContext:conversation.managedObjectContext];
-}
+//+ (ZMClientMessage *)appendSelfConversationWithClearedOfConversation:(ZMConversation *)conversation
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation.m:762]: Should be deleted?

File: Source/Utilis/Protos/GenericMessage+Helper.swift [hunk 4]
```diff
@@ -490,6 +490,13 @@ extension WireProtos.MessageEdit: MessageCapable {
 }
 
 extension Cleared: MessageCapable {
+    public init(timeStamp: Date, conversationID: UUID) {
```
Comment (typfel) [inline Source/Utilis/Protos/GenericMessage+Helper.swift:493]:
```suggestion
    public init(timestamp: Date, conversationID: UUID) {
```

File: Source/Model/Conversation/ZMConversation+Message.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation+Message.swift]: Done

File: Source/Model/Conversation/ZMConversation.m (hunk 2 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation.m]: Done

File: Source/Model/Conversation/ZMConversation.m (hunk 3 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation.m]: Done

**Pull request merged**